pallets/dca/src/lib.rs

```diff
@@ -911,13 +915,30 @@ where
 	fn find_next_free_block(blocknumber: T::BlockNumber) -> Result<T::BlockNumber, DispatchError> {
 		let mut next_execution_block = blocknumber;
 
-		// In a bound fashion, we search for next free block with the delays of 1 - 2 - 4 - 8 - 16.
-		for retry_index in 0u32..=RETRY_TO_SEARCH_FOR_FREE_BLOCK {
+		let mut randomness_generator =
+			Self::get_randomness_generator(frame_system::Pallet::<T>::current_block_number());
+
+		let radius_ranges = [
```

Comment:
you dont have to store this in code, it is simple loop where ranges are `(2^i, 2^i+1 - 1)`